src/Workspaces/MSBuildTest/OpenProjectsTests.cs

```diff
@@ -0,0 +1,222 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.MSBuild.UnitTests
+{
+    // Generate a list of templates by running `dotnet new list --type project`
```

Comment:
Is the list in the tests just all of them, or did you filter or remove them further? Because we could just make the discovery of the template list dynamic and test everything...

Reply:
Verified that windows templates are not offered on MacOS.

Reply:
Moved to generating this list.